### PR TITLE
Move Parallax helper to use prototypes.

### PR DESCRIPTION
**Pull request by nwhite89**
- Adds unit tests for testing the Parallax helper
- Changes the parallax helper to be created via an instance

TP: https://rockabox.tpondemand.com/entity/10381

### CODE REVIEW

File: src/parallax-scrolling.js [hunk 1]
```diff
@@ -15,67 +15,209 @@ define([
...
+        // Whether or not to use inverted scrolling direction (defaulting to normal direction)
+        this.invert = invert || false;
+
+        var $this = this,
+            offsetTop = this._offset(ele.parentNode).y,
+            scrollDistance = (eleHeight - viewableHeight);
+
+        // The handler in which to be used for firing when scrolling
+        return function () {
+            var winPositon = $this._getWindowPositions(win),
```
Comment (jonman) [inline src/parallax-scrolling.js]: `winPosition`

File: src/parallax-scrolling.js [hunk 2]
```diff
@@ -15,67 +15,209 @@ define([
...
      *
-     * @return {Number}            The positioning of the element (in pixels)
+     * // When the window is scrolled we want to fire the parallax handler
+     * events.addListener(window, 'scroll', function () {
+     * 	parallaxHelper();
+     * });
+     *
+     * // When the window is resized we want to fire the parallax helper
+     * events.addListener(window, 'resize', function () {
+     * 	parallaxHelper();
```
Comment (jonman) [inline src/parallax-scrolling.js]: This should `events.addListener(window, 'resize', parallaxHelper);` or
```js
events.addListener(window, 'resize', function (event) {
    parallaxHelper(event);
});
```

File: src/parallax-scrolling.js [hunk 3]
```diff
@@ -15,67 +15,209 @@ define([
...
+        win = win || window;
+        // Whether or not to use inverted scrolling direction (defaulting to normal direction)
+        this.invert = invert || false;
+
+        var $this = this,
+            offsetTop = this._offset(ele.parentNode).y,
+            scrollDistance = (eleHeight - viewableHeight);
+
+        // The handler in which to be used for firing when scrolling
+        return function () {
```
Comment (jonman) [inline src/parallax-scrolling.js]: Think this should retain the `event` object fired from the `onscroll` event.

File: src/parallax-scrolling.js [hunk 4]
```diff
@@ -15,67 +15,209 @@ define([
...
+     * inverted or not).
+     *
+     * @memberOf module:parallax-scrolling
+     *
+     * @private
+     *
+     * @param  {Number} offsetTop The offsetTop offset of the element.
+     * @param  {Number} scrollTop The offset top of the current viewport compared to the window.
+     * @param  {Number} distance  The distance between the windows height and the viewable height.
+     * @param  {Boolean} invert   Whether or not to inver the scrolling direction.
```
Comment (jonman) [inline src/parallax-scrolling.js]: `or not to invert the scrolling`

File: src/parallax-scrolling.js [hunk 5]
```diff
@@ -15,67 +15,209 @@ define([
...
+     * Get the value for the positional value for setting to show the top of the content
+     *
+     * @memberOf module:parallax-scrolling
+     *
+     * @private
+     *
+     * @return {Number} The positional value for the content showing at the top of the element.
+     */
+    ParallaxScrolling.prototype._positionTop = function () {
+        return '0px';
```
Comment (jonman) [inline src/parallax-scrolling.js]: Is this always return `0px`?

File: src/parallax-scrolling.js (hunk 2 above)
Comment (nwhite89) [inline src/parallax-scrolling.js]: ``parallaxHelper`` doesn't need ``event``

File: src/parallax-scrolling.js (hunk 5 above)
Comment (nwhite89) [inline src/parallax-scrolling.js]: should always return 0 yes

File: src/parallax-scrolling.js (hunk 3 above)
Comment (nwhite89) [inline src/parallax-scrolling.js]: as above ``event`` not needed for the handler

**Pull request merged**